### PR TITLE
testing: Finalize testing and full cleanup of mpconfigport.h.

### PR DESCRIPTION
**Pull request by bargfred**

### CODE REVIEW

File: ports/psoc6/mpconfigport.h [hunk 1]
```diff
@@ -3,7 +3,7 @@
  *
  * The MIT License (MIT)
  *
- * Copyright (c) 2020-2021 Damien P. George
+ * Copyright (c) 2022-2023 Infineon Technologies AG
```
Comment (jaenrig-ifx) [inline ports/psoc6/mpconfigport.h:6]: I think we should keep the former copyright and add our own

File: ports/psoc6/mpconfigport.h [hunk 2]
```diff
@@ -26,21 +26,23 @@
 
 // Options controlling how MicroPython is built, overriding defaults in py/mpconfig.h
 #include <stdint.h>
-#include "mpconfigboard.h"
```
Comment (jaenrig-ifx) [inline ports/psoc6/mpconfigport.h:29]: Why is this removed? With this include we can add board configuration, and provide here default port configuration, but prevent to overwrite that the board with #ifdef/#ifndef.

Comment (jaenrig-ifx) [reply ports/psoc6/mpconfigport.h:29]: Also if the flag is coming from the compiler

File: ports/psoc6/mpconfigport.h [hunk 3]
```diff
@@ -86,139 +84,66 @@
...
+#define MICROPY_MAKE_POINTER_CALLABLE(p)        ((void *)((mp_uint_t)(p) | 1))
 
 #define MP_SSIZE_MAX (0x7fffffff)
-typedef intptr_t mp_int_t; // must be pointer size
+typedef intptr_t mp_int_t;  // must be pointer size
 typedef uintptr_t mp_uint_t; // must be pointer size
 typedef intptr_t mp_off_t;
 
 // We need to provide a declaration/definition of alloca()
 #include <alloca.h>
```
Comment (boramonideep) [inline ports/psoc6/mpconfigport.h:132]: Any special reason for including this header? Why not at the top with stdint?

File: ports/psoc6/mpconfigport.h [hunk 4]
```diff
@@ -86,139 +84,66 @@
...
-#define MICROPY_ALLOC_PARSE_CHUNK_INIT    (16)
-#define MICROPY_KBD_EXCEPTION             (1)
-
-#define MICROPY_ENABLE_SOURCE_LINE        (1)
 
-#define MICROPY_USE_INTERNAL_PRINTF       (0)
-#define MICROPY_ENABLE_FINALISER          (1)
-#define MICROPY_VFS_POSIX                 (0)
-#define MICROPY_READER_POSIX              (0)
+// set in Makefile
```
Comment (boramonideep) [inline ports/psoc6/mpconfigport.h:137]: Could we remove if already set in Makefile?

**Review by jaenrig-ifx — APPROVED** [review]
👍 I cannot judge all the options enablement. The board configuration is not important right now, but we should keep the hierarchy of configuration, which makefile boards can include certain preprocessor defines, then the mpconfigboard.h and then the mpconfigport.h. 
There is also a default config which basically take default mpy values, which in case of features that we are not maybe having much clue on how they work, they could take that value.

File: ports/psoc6/mpconfigport.h [hunk 5]
```diff
@@ -227,15 +152,17 @@ typedef intptr_t mp_off_t;
 
 #include "shared/runtime/interrupt_char.h"
```
Comment (boramonideep) [inline ports/psoc6/mpconfigport.h:153]: Similar to above, couldn't we move the header to the top?

File: tests/psoc6/README [hunk 6]
```diff
@@ -217,7 +217,7 @@ run-tests.py         : run tests from host platform
```
Comment (boramonideep) [inline tests/psoc6/README:217]: Maybe at some point we could dynamically display how many and which tests passed/failed in the CI/CD flow. Similar to README badges that are available nowadays. Just an idea. @jaenrig-ifx @bargfred

File: ports/psoc6/mpconfigport.h [hunk 7]
```diff
@@ -86,139 +84,66 @@
...
-#define MICROPY_READER_POSIX              (0)
+// set in Makefile
+// #define MICROPY_QSTR_EXTRA_POOL                 mp_qstr_frozen_const_pool
 
+#define MICROPY_USE_INTERNAL_PRINTF             (0)
 
-#define MICROPY_REPL_INFO                 (1)
+#define MICROPY_REPL_INFO                       (1)
 
 // Set to zero explicitly ! Or mounting the /flash fs must be done inside try/except for both branches !
```
Comment (boramonideep) [inline ports/psoc6/mpconfigport.h:144]: Maybe already set elsewhere or is redundant and can be removed?

**Review by boramonideep — APPROVED** [review]
All good. Thanks for the work @bargfred . 

Just a few questions and suggestions in comments.

Also, the file `tests/psoc6/run_psoc6_tests.sh` seems to be empty which could be removed?

File: ports/psoc6/mpconfigport.h (hunk 2 above)
Comment (boramonideep) [reply ports/psoc6/mpconfigport.h:29]: I guess its just moved to the end of the file so that nothing board-specific is overwritten.

File: ports/psoc6/mpconfigport.h (hunk 1 above)
Comment (bargfred) [reply ports/psoc6/mpconfigport.h:6]: Well, our file bears no resemblance to the rp2 / minimal port files, so copyright should be IFX. Why should it be D. George ?

File: ports/psoc6/mpconfigport.h (hunk 2 above)
Comment (bargfred) [reply ports/psoc6/mpconfigport.h:29]: Right, it has been moved to the end such that board specific changes overwrite port specific ones. Alternatively, all defines have to be enclosed in #ifndef ... #endif, which is much more messy than this way.

File: ports/psoc6/mpconfigport.h (hunk 3 above)
Comment (bargfred) [reply ports/psoc6/mpconfigport.h:132]: Good point. Put here in order to be close to the code that makes use of it, but maybe better at the top => done.

File: ports/psoc6/mpconfigport.h (hunk 4 above)
Comment (bargfred) [reply ports/psoc6/mpconfigport.h:137]: done.

File: ports/psoc6/mpconfigport.h (hunk 7 above)
Comment (bargfred) [reply ports/psoc6/mpconfigport.h:144]: It was more of a reminder for me, removed it.

File: ports/psoc6/mpconfigport.h (hunk 5 above)
Comment (bargfred) [reply ports/psoc6/mpconfigport.h:153]: done

File: ports/psoc6/mpconfigport.h (hunk 2 above)
Comment (jaenrig-ifx) [reply ports/psoc6/mpconfigport.h:29]: okay... as long as the defines are properly overwritten by the board configuration.

File: tests/psoc6/README (hunk 6 above)
Comment (bargfred) [reply tests/psoc6/README:217]: Not sure what you mean exactly, but let us discuss on the phone.

File: ports/psoc6/mpconfigport.h (hunk 2 above)
Comment (bargfred) [reply ports/psoc6/mpconfigport.h:29]: Leaving it this way for now, if decided to to it differently we could change it as we incorporate the WIFI changes.

File: ports/psoc6/mpconfigport.h (hunk 1 above)
Comment (bargfred) [reply ports/psoc6/mpconfigport.h:6]: I think we have to review ALL files before providing the port officially, because they all should have a common header / disclaimer as in the other ports. And many of our files do not.

Comment (jaenrig-ifx) [reply ports/psoc6/mpconfigport.h:6]: as it was based on the template and went over several iterations of changes... i guess it was an initial file with a copyright which we modified, and always public. But I am not a lawyer, and I don´t think this would create any legal problem anyhow :)

File: tests/psoc6/README (hunk 6 above)
Comment (boramonideep) [reply tests/psoc6/README:217]: image

https://www.runforesight.com/blog/top-5-badges-that-will-show-your-github-repository-is-well-tested-trusted

Something similar to this.

Comment (jaenrig-ifx) [reply tests/psoc6/README:217]: Yes, that is a good idea. That is already available for github actions. There is a badge in the readme for unix, but not for the rest of the ports.

Comment (bargfred) [reply tests/psoc6/README:217]: The diff is empty, not the file.

A badge would be nice.
